### ISSUE TITLE
Does Pytorch have the function that can obtain sub-matrix according to index? 

### ISSUE THREAD

**Issue by zxyyxzz** [pytorch/pytorch#49278]
## 🚀 Feature

If i have a multidimensional tensor: **matrix = 500\*600, 500, 600:** [batch, H, W]

Now i want a function which can obtain a sub-matrix(h, w) of the matrix on H, W dim according to the index in the first dim:
for example:

```python
     kernel_H_W= 12
     all_sub-matrix = []
     for local_index in range(matrix .shape[0]):
            row = local_index // w
            col = local_index % w

            sub-matrix= matrix [local_index, row:row + 2 * kernel_H_W+ 1,col:col + 2 * kernel_H_W+ 1].contiguous()
            all_sub-matrix.append(sub-matrix.unsqueeze(0))
     all_sub-matrix = torch.cat(all_sub-matrix, dim =0) 
```

## Motivation

if i use the above code, the speed is so slow, is there any function in Pytorch can obtain the sub-matrix?

**Comment by zxyyxzz** [comment]
@xwang233
